backend/src/org/akvo/dash/endpoint/dataset.clj

```diff
@@ -1,131 +1,97 @@
-                                :datasource   (:id datasource)}
-                dataset_meta   {:id           (squuid)
-                                :dataset_name (get-in req [:body "name"])
-                                :dataset      (:id dataset)}
-                import         {:datasource (:id datasource)}
-                transformation {:id      (squuid)
-                                :dataset (:id dataset)
-                                :fns     {:fns []}}
-                res            (clojure.java.jdbc/with-db-transaction [tx db]
-                                 {:s   (insert-datasource tx datasource)
-                                  :ds  (insert-dataset tx dataset)
-                                  :dsm (insert-dataset_meta tx dataset_meta)
-                                  :t   (insert-transformation tx transformation)
-                                  :i   (insert-import tx import)})
-                resp           {"id"   (:id dataset)
-                                "name" (:dataset_name dataset_meta)}]
-            (if (= "DATA_FILE" (get-in datasource [:spec "kind"]))
-              (do
-                (import/do-import db
-                                  datasource
-                                  (-> res :i :id))
-                (rr (assoc resp "status" "OK")))
-              (do
-                (scheduling/schedule #(import/job db
-                                                  (-> res :i :id)))
-                (rr (assoc resp "status" "PENDING")))))
+          (handle-datasets-POST! request
+                                 (connection tm tenant))
           (catch Exception e
-            (rr {:error "Could not complete upload"
-                 :status "FAILED"})))))
+            (pprint e)
+            (pprint (.getNextExcpetion e))
+            (rr {:error  "Could not complete upload."
+                 :status "FAILURE"})))))
 
 
     (context "/:id" [id]
 
       (GET "/" []
-        (let [r (dataset-by-id db
-                               {:id (str->uuid id)})]
-          (if (nil? r)
-            (rr "404 Not Found"
-                {:status  404
-                 :headers {"content-type" "text"}})
-            (rr
-             (let [k        [:id :datasource :name :status :created :modified
-                             :transformations]
-                   resp-map (select-keys r k)]
-               (condp = (:status r)
-                 "OK"      (assoc resp-map
-                                  :columns (json/parse-string
-                                            (slurp (get-in r [:noble "noble"]))))
-                 "FAILURE" (assoc resp-map
-                                  :reason
-                                  "Could not find dataset on remote server (404)")
-                 resp-map)))))) ;; "PENDING" do nothing
+        (fn [{tenant :tenant :as request}]
+          (let [{:keys [:id :d :name :status :created :modified] :as ds}
```

Comment:
The dataset "data". I'm guessing all this will go away with the new dataset implementation.